src/map/clan.c

```diff
@@ -0,0 +1,839 @@
+
+	CREATE(c, struct clan, 1);
+
+	c->clan_id = id;
+	safestrncpy(c->constant, aConst, NAME_LENGTH);
+	safestrncpy(c->name, aName, NAME_LENGTH);
+	safestrncpy(c->master, aLeader, NAME_LENGTH);
+	safestrncpy(c->map, aMap, MAP_NAME_LENGTH_EXT);
+	c->max_member = clan->max;
+	if (c->max_member > MAX_CLAN) {
+		ShowWarning("Clan %d:%s specifies higher capacity (%d) than MAX_CLAN (%d)\n", c->clan_id, c->name, c->max_member, MAX_CLAN);
+		c->max_member = MAX_CLAN;
+	}
+	c->connect_member = 0;
+
+	if (libconfig->setting_lookup_int(clan_, "KickTime", &kicktime)) {
+		c->kick_time = 60 * 60 * kicktime;
+	} else {
+		c->kick_time = clan->kicktime;
+	}
+	if (libconfig->setting_lookup_int(clan_, "CheckTime", &kickchecktime)) {
+		c->check_time = 60 * 60 * max(1, kickchecktime) * 1000;
+	} else {
+		c->check_time = clan->checktime;
+	}
+
+
+	if ((buff = libconfig->setting_get_member(clan_, "Buff")) != NULL) {
+		struct clan_buff *b = &c->buff;
+		const char *str = NULL;
+
+		if (!libconfig->setting_lookup_int(buff, "Icon", &b->icon)) {
+			const char *temp_str = NULL;
+			if (libconfig->setting_lookup_string(buff, "Icon", &temp_str)) {
+				if (*temp_str) {
+					if (!script->get_constant(temp_str, &b->icon)) {
+						ShowWarning("Clan %d: Invalid Icon Value. Defaulting to SI_BLANK.\n", c->clan_id);
+						b->icon = -1;
+					}
+				}
+			}
+		}
+
+		if (libconfig->setting_lookup_string(buff, "Script", &str)) {
+			b->script = *str ? script->parse(str, source, -b->icon, SCRIPT_IGNORE_EXTERNAL_BRACKETS, NULL) : NULL;
+		}
+	}
+
+	if (allies != NULL) {
+		const char *ally_const;
+
+		for (i = 0; i < libconfig->setting_length(allies) &&  i < clan->maxalliances; i++) {
+			if ((ally_const = libconfig->setting_get_string_elem(allies, i)) != NULL)
+				safestrncpy(c->allies[i].constant, ally_const, NAME_LENGTH);
+		}
+	}
+	if (antagonists != NULL) {
+		const char *antg_const;
+
+		for (i = 0; i < libconfig->setting_length(antagonists) && i < clan->maxalliances; i++) {
+			if ((antg_const = libconfig->setting_get_string_elem(antagonists, i)) != NULL)
+				safestrncpy(c->antagonists[i].constant, antg_const, NAME_LENGTH);
+		}
+	}
+	clan->read_db_additional_fields(c, clan_, n, source);
+	idb_put(clan->db, c->clan_id, c);
+	c->tid = timer->add(timer->gettick() + c->kick_time, clan->inactivity_kick, c->clan_id, 0);
+
+	return true;
+}
+
+/**
+ * Reads Clan DB included by clan configuration file.
+ *
+ * @param settings The Settings Group from config file.
+ * @param source File name.
+ */
+void clan_read_db(struct config_setting_t *settings, const char *source)
+{
+	struct config_setting_t *clans;
+
+	nullpo_retv(settings);
+
+	if ((clans = libconfig->setting_get_member(settings, "clans")) != NULL ) {
+		int i;
+		int clan_count = libconfig->setting_length(clans);
+		for (i = 0; i < clan_count; i++) {
+			clan->read_db_sub(clans, i, source);
+		}
+		ShowStatus("Done reading '" CL_WHITE "%d" CL_RESET "' entries in '" CL_WHITE "%s" CL_RESET "'.\n", clan_count, source);
```

Comment:
There is a possibility of `clan_count` being `0`. Which would call on both `ShowStatus` and `clan->constants()` to do some useless if not harmful work.